cypress/support/data_generation/dataForSpecificTest.js

```diff
@@ -0,0 +1,59 @@
+function generateInvalidEmailForTest(testTitle, emails) {
+  const {
+    noNamePartEmail,
+    noAtSymbolEmail,
+    noDomainPartEmail,
+    noDotPartEmail,
+    noTopDomainPartEmail
+  } = emails;
+
+  if (testTitle.includes(`email without [name] part`)) {

```

Comment:
Now I understand why you need wrapping testTitles. It is interesting approach, but it adds complexity.
1. It is not clear while reading tests from where we get needed test data
2. It is hard to maintain tests, since after changing test title it will stop working

Reply:
This is my first project in the automated E2E testing. And I'm very proud of it that I made it by myself. And this approach my own idea, I have made it by myself :)

I can say, that there is an opportunity to know from where we get needed test data. Inside the methods where I used this function to return invalid email there are params which indicate where the data will come from. But from the point of view of a maintenance, I'm agree that this is a bit thoughtless approach. But for now I think it works well :)